map/mwm_url.cpp

```diff
@@ -100,54 +88,50 @@ bool ParseLatLon(std::string const & key, std::string const & value,
 }
 }  // namespace
 
-void ParsedMapApi::SetBookmarkManager(BookmarkManager * manager)
+template <size_t S> size_t findPrefix(string const & url, const std::array<std::string_view, S> prefixes)
 {
-  m_bmManager = manager;
+  for (std::string_view prefix : prefixes)
+  {
+    if (strings::StartsWith(url, prefix))
+      return prefix.size();
+  }
+  return string::npos;
 }
 
-ParsedMapApi::ParsingResult ParsedMapApi::SetUrlAndParse(string const & url)
+ParsedMapApi::UrlType ParsedMapApi::SetUrlAndParse(string const & raw)
 {
   Reset();
 
-  auto const u = url::Url(url);
-  auto const urlType = GetUrlType(u);
-  m_isValid = Parse(u, urlType);
+  SCOPE_GUARD(guard, [this]() { Reset(); });
 
-  return {urlType, m_isValid};
-}
+  // kGe0Prefixes supports API and short links, kLegacyMwmPrefixes - only API.
+  const size_t newPrefix = findPrefix(raw, ge0::kGe0Prefixes);
+  const size_t anyPrefix = newPrefix != string::npos ? newPrefix : findPrefix(raw, ge0::kLegacyMwmPrefixes);
```

Comment:
nit: Make a function like
```
std::pair<size_t, size_t> FindGe0AndMwmPrefixes()
```
in parser.cpp scope to avoid external exposing of kGe0Prefixes and kLegacyMwmPrefixes arrays.
Then just:
```
auto const [newPrefix, anyPrefix] = FindGe0AndMwmPrefixes(raw);
```

Reply:
kGe0Prefixes are used by ge0 itself.

Reply:
Yes, my point is to keep arrays in ge0/parser.cpp only and expose a FindGe0AndMwmPrefixes function.

Reply:
I reworked it in a different way.